quickwit-serve/src/lib.rs

```diff
@@ -115,9 +124,35 @@ pub async fn serve_cli(args: ServeArgs) -> anyhow::Result<()> {
         .next()
         .with_context(|| "No index available.")?
         .to_string();
-    let search_service_impl = SearchServiceImpl::new(metastore_router, storage_resolver);
-    let rest_routes = rest::search_handler(Arc::new(search_service_impl));
+
+    let host_key = read_host_key(args.host_key_path.as_path())?;
+    let swim_addr = http_addr_to_swim_addr(args.rest_socket_addr);
+    let cluster = Arc::new(Cluster::new(host_key, swim_addr)?);
+    for peer_socket_addr in args.peer_socket_addrs {
```

Comment:
Contrary to what I told you in a preview CR, using `to_socket_addr` could actually be a nice thing here.
We could actually have to_socket_addrs return all of the ip addrs and add 3 of them as peer node.

That way in a kubernetes settingfor instance, someone could create a headless service and the peer nodes would be discovered via DNS.